src/components/timeline/TrackCounterGraph.js

```diff
@@ -99,20 +112,25 @@ class TrackMemoryCanvas extends React.PureComponent<CanvasProps> {
       // Draw the chart.
       const rangeLength = rangeEnd - rangeStart;
       ctx.lineWidth = deviceLineWidth;
-      ctx.strokeStyle = ORANGE_50;
-      ctx.fillStyle = '#ff940088'; // Orange 50 with transparency.
+      ctx.strokeStyle = graphStrokeColor;
+      ctx.fillStyle = graphFillColor; // Blue 50 or Orange 50 with transparency.
       ctx.beginPath();
-
       // The x and y are used after the loop.
       let x = 0;
       let y = 0;
+      let yOld = 0;
       for (let i = 0; i < samples.length; i++) {
         // Create a path for the top of the chart. This is the line that will have
         // a stroke applied to it.
         x = (deviceWidth * (samples.time[i] - rangeStart)) / rangeLength;
+        // Y position for visual progress graph calculate based on sample count
+        // instead of accumulated count.
+        const unitGraphCount =
+          graphType === GRAPH_TYPE.visualProgress
```

Comment:
Instead of doing an if/else check each iteration of this loop, I would add a `prop` value that can inject a function that will do this.

`computeGraphHeight: IndexIntoSamplesTable => DevicePixels`